update test_rpc

tktrungna opened the pull request:
Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62838
* #62797
* #62796
* #62795
* __->__ #62781

Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

tktrungna committed:
update test_rpc

[ghstack-poisoned]

tktrungna committed:
Update on "update test_rpc"




[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124391).

tktrungna committed:
Update on "update test_rpc"


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124391).

tktrungna committed:
Update on "update test_rpc"


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_rpc"


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_rpc"


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_rpc"


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124391).

janeyx99 commented:
I added the master workflow just in case--especially as it seems test_rpc is only run for non-shard jobs, and the nonsharded jobs we have on PRs are CPU only.

janeyx99 commented:
https://app.circleci.com/pipelines/github/pytorch/pytorch/361849/workflows/b7641fd8-dc2e-4bec-9d6c-b9e146b91ca1/jobs/15263375 error seems legit

tktrungna committed:
Update on "update test_rpc"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124391).

tktrungna committed:
Update on "update test_rpc"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna committed:
Update on "update test_rpc"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

zhouzhuojie commented on .jenkins/pytorch/test.sh:
```diff
@@ -315,9 +316,12 @@ test_rpc() {
     echo "Testing RPC C++ tests"
     # NB: the ending test_rpc must match the current function name for the current
     # test reporting process (in print_test_stats.py) to function as expected.
+    ln -sf "$TORCH_LIB_DIR"/libtorch* "$TORCH_BIN_DIR"
+    ln -sf "$TORCH_LIB_DIR"/libc10* "$TORCH_BIN_DIR"
+    ln -sf "$TORCH_LIB_DIR"/libtbb* "$TORCH_BIN_DIR"
```
Wondering if there's way that we can notify people if they build more binaries that need to be included for test

tktrungna replied:
I am struggling with this part too, as adding sym links doesn't seem to be an intuitive way to write test (as one may expect the executable file under installed `bin` to find the shared library in `lib`). Not sure this one https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/RPATH-handling is a good reference but I will create an issue to follow up this part

zhouzhuojie reviewed (approved):
lgtm, CI is green

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124391).

tktrungna committed:
Update on "update test_rpc"

Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass


Differential Revision: [D30124391](https://our.internmc.facebook.com/intern/diff/D30124391)

[ghstack-poisoned]

tktrungna commented:
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30124391).
